examples/collinear_magnetism.jl

```diff
@@ -0,0 +1,88 @@
+# # Collinear spin and magnetic systems
+#
+# In this example we consider iron in the BCC phase.
+# To show that this material is ferromagnetic we will model it once
+# allowing collinear spin polarization and once without
+# and compare the resulting SCF energies. Since in ferromagnetic
+# materials the ground state features aligned spins,
+# i.e. a spin-polarized density, the true ground state can only
+# be found if collinear spins are allowed.
+#
+# First we setup BCC iron without spin polarization.
+# For this purpose we construct a standard BCC lattice and place
+# a single iron atom inside the cell.
+using DFTK
+
+a = 5.42352  # Bohr
+lattice = a / 2 * [[-1  1  1];
+                   [ 1 -1  1];
+                   [ 1  1 -1]]
+Fe = ElementPsp(:Fe, psp=load_psp("hgh/lda/Fe-q8.hgh"))
+atoms = [Fe => [zeros(3)]];
+
+# To get the ground-state energy we use an LDA model and rather moderate
+# discretisation parameters.
+
+kgrid = [3, 3, 3]  # k-point grid (Regular Monkhorst-Pack grid)
+Ecut = 15          # kinetic energy cutoff in Hartree
+model_nospin = model_LDA(lattice, atoms, temperature=0.01)
+basis_nospin = PlaneWaveBasis(model_nospin, Ecut; kgrid=kgrid)
+
+scfres_nospin = self_consistent_field(basis_nospin, tol=1e-6, mixing=KerkerMixing())
+
+scfres_nospin.energies
+
+# Since did not specify any initial magnetic moment on the iron atom,
+# DFTK will automatically assume that a calculation with only spin-paired
+# electrons should be performed. As a result the obtained ground state
+# features no spin-polarization.
+
+# Now we repeat the calculation, but give the iron atom an initial magnetic moment.
+# For specifying the magnetic moment DFTK uses units of ``\frac{μ_B}{2}``, which
```

Comment:
Well that's the thing ... `\mu_B` is not atomic units ... in atomic units `\mu_B` is actually 1/2, but it's more natural to use units of `\mu_B` (not /2 ... that's a typo) for this because than the magnetisation is just the number of excess electrons. I think this is pretty confusing and clearly deserves to be explained.